src/java/org/apache/cassandra/service/accord/serializers/DepsSerializers.java

```diff
@@ -74,164 +83,294 @@ public DepsSerializer(UnversionedSerializer<Range> tokenRange)
+                if (xCount == 1)
+                {
+                    xtoy[0] = xtoy.length;
+                    for (int i = 0 ; i < yCount ; ++i)
+                        xtoy[1 + i] = i;
+                }
+                else if (yCount == 1)
+                {
+                    for (int i = 0 ; i < xCount ; ++i)
+                        xtoy[i] = xCount + i + 1;
+                }
+                else if (yCount == 0)
+                {
+                    for (int i = 0 ; i < xCount ; ++i)
+                        xtoy[i] = xCount;
+                }
+                else
+                {
+                    Invariants.require(length == 0);
+                }
+            }
+            else
+            {
+                deserializePackedInts(xtoy, 0, xCount, xtoy.length, in);
+                deserializePackedInts(xtoy, xCount, xtoy.length, yCount - 1, in);
+            }
+            return xtoy;
+        }
+
+        private static void deserializePackedInts(int[] vs, int from, int to, int max, DataInputPlus in) throws IOException
+        {
+            int bitsPerEntry = BitUtils.numberOfBitsToRepresent(max);
+            int mask = -1 >>> (32 - bitsPerEntry);
+            int remainingBytes = (bitsPerEntry * (to - from) + 7) / 8;
+            long buffer = 0L;
+            int bufferCount = 0;
+            for (int i = from; i < to; i++)
+            {
+                int v = (int)buffer & mask;
+                if (bufferCount >= bitsPerEntry)
+                {
+                    bufferCount -= bitsPerEntry;
+                    buffer >>>= bitsPerEntry;
+                }
+                else
+                {
+                    int newBufferCount;
+                    if (remainingBytes >= 8)
+                    {
+                        buffer = in.readLong();
+                        newBufferCount = 64;
+                        remainingBytes -= 8;
+                    }
+                    else
+                    {
+                        Invariants.require(remainingBytes > 0);
+                        newBufferCount = remainingBytes * 8;
+                        buffer = in.readLeastSignificantBytes(remainingBytes);
+                        remainingBytes = 0;
+                    }
+                    int readExtra = bitsPerEntry - bufferCount;
+                    int extraBits = (int)buffer & (mask >>> bufferCount);
+                    v |= extraBits << bufferCount;
+                    bufferCount = newBufferCount - readExtra;
+                    buffer >>>= readExtra;
+                }
+                Invariants.require(v <= max);
+                vs[i] = v;
+            }
+        }
+
         @Override
         public long serializedSize(D deps)
         {
-            long size;
+            boolean keysByTxnId = forceByTxnId || deps.keyDeps.hasByTxnId();
+            boolean rangesByTxnId = forceByTxnId || deps.rangeDeps.hasByTxnId();
+            long size = 1;
```

Comment:
Is it guaranteed to be 1 byte? Looks like we're writing vint there, maybe add a comment here, or write it as byte?